pkg/state/kubernetes.go

```diff
@@ -0,0 +1,242 @@
+	}, nil
+}
+
+// getKubernetesConfig returns a Kubernetes REST config, trying in-cluster first,
+// then falling back to out-of-cluster configuration using kubeconfig
+func getKubernetesConfig() (*rest.Config, error) {
+	// Try in-cluster config first
+	config, err := rest.InClusterConfig()
+	if err == nil {
+		return config, nil
+	}
+
+	// If in-cluster config fails, try out-of-cluster config
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	config, err = kubeConfig.ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kubernetes config (tried both in-cluster and out-of-cluster): %w", err)
+	}
+
+	return config, nil
+}
+
+// getCurrentNamespace determines the current Kubernetes namespace
+func getCurrentNamespace() string {
+	// Try to read from service account namespace file (in-cluster)
+	if data, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
+		return strings.TrimSpace(string(data))
+	}
+
+	// Try to read from kubeconfig context
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+
+	rawConfig, err := kubeConfig.RawConfig()
+	if err != nil {
+		// Fallback to default namespace
+		return "default"
+	}
+
+	if ctxConfig := rawConfig.Contexts[rawConfig.CurrentContext]; ctxConfig != nil && ctxConfig.Namespace != "" {
+		return ctxConfig.Namespace
+	}
+
+	// Default fallback
+	return "default"
+}
+
+// getConfigMapName returns the ConfigMap name for a given runconfig name
+func (k *KubernetesStore) getConfigMapName(name string) string {
+	return fmt.Sprintf("%s%s-%s", ConfigMapPrefix, k.storeName, name)
+}
+
+// GetReader returns a reader for the state data from a ConfigMap
+func (k *KubernetesStore) GetReader(ctx context.Context, name string) (io.ReadCloser, error) {
+	configMapName := k.getConfigMapName(name)
+
+	configMap, err := k.client.CoreV1().ConfigMaps(k.namespace).Get(ctx, configMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ConfigMap %s: %w", configMapName, err)
+	}
+
+	data, exists := configMap.Data[ConfigMapDataKey]
+	if !exists {
+		return nil, fmt.Errorf("runconfig data not found in ConfigMap %s", configMapName)
+	}
+
+	return io.NopCloser(strings.NewReader(data)), nil
+}
+
+// GetWriter returns a writer for the state data to a ConfigMap
+func (k *KubernetesStore) GetWriter(ctx context.Context, name string) (io.WriteCloser, error) {
+	return &configMapWriter{
+		store:  k,
+		ctx:    ctx,
+		name:   name,
+		buffer: &bytes.Buffer{},
+	}, nil
+}
+
+// Delete removes the ConfigMap for the given name
+func (k *KubernetesStore) Delete(ctx context.Context, name string) error {
```

Comment:
Not sure why we want to delete the CM here, I would have thought the Operator would have been the one to do it?